include/snowflake/ClientConfigParser.hpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2024 Snowflake Computing
+ */
+
+#ifndef SNOWFLAKE_CONFIGPARSER_HPP
+#define SNOWFLAKE_CONFIGPARSER_HPP
+
+#include <string>
+#include "client_config_parser.h"
+
+namespace Snowflake
+{
+namespace Client
+{
+  struct ClientConfigException : public std::exception
```

Comment:
Why do we need a class if it has no member variables?

Reply:
Just using it as a wrapper for exception to pass along the error message. I saw other classes having a similar exception struct, but I can just make the function return false instead of throwing an exception if you prefer.